packages/firebase_database/firebase_database/example/test_driver/query_e2e.dart

```diff
@@ -46,19 +49,24 @@ void runQueryTests() {
     });
 
     test('correct order returned from query', () async {
-      Event event = await database
+      final c = Completer<List<Map<String, dynamic>>>();
+      final items = <Map<String, dynamic>>[];
+
+      // ignore: unawaited_futures
+      database
           .reference()
           .child('ordered')
           .orderByChild('value')
-          .onValue
-          .first;
+          .onChildAdded
```

Comment:
So onValue does not return the results in correct order

Reply:
no, `onValue` receives a snapshot with all items in an unordered map